netconn.go

```diff
@@ -40,10 +40,14 @@ import (
 func openConnection(uri *url.URL, tlsc *tls.Config, timeout time.Duration, headers http.Header, websocketOptions *WebsocketOptions, dialer *net.Dialer) (net.Conn, error) {
 	switch uri.Scheme {
 	case "ws":
-		conn, err := NewWebsocket(uri.String(), nil, timeout, headers, websocketOptions)
+		dialURI := uri // #623 - Gorilla Websockets does not accept URL's where uri.User != nil
+		uri.User = nil
+		conn, err := NewWebsocket(dialURI.String(), nil, timeout, headers, websocketOptions)
```

Comment:
Unless I'm mistaken, this code alters the original url (especially since it was passed in as a pointer to this function).

Shouldn't it be like this:
```go
		dialURI := *uri // #623 - Gorilla Websockets does not accept URL's where uri.User != nil
		dialURI.User = nil
		conn, err := NewWebsocket(dialURI.String(), nil, timeout, headers, websocketOptions)
```

If so, then the same issue is on the `"wss"` branch below

Reply:
ping @MattBrittan (apologies if inappropriate, but doing this since I'm responding to a closed PR)

Reply:
Well spotted; this change did work (This was a quick fix, made on a customers site, that I needed to move to websockets urgently to bypass a firewall :-) ) but I can now see that it would have failed on subsequent connections (funnily enough I did have an issue with that site but put it down to websockets :-) ). I'll push a patch through shortly.